app/lib/actions/katello/applicability/hosts/bulk_generate.rb

```diff
@@ -10,7 +10,14 @@ class BulkGenerate < Actions::EntryAction
           def run
             input[:host_ids].each do |host_id|
               content_facet = ::Host.find(host_id).content_facet
-              content_facet.calculate_and_import_applicability
+            rescue ActiveRecord::RecordNotFound
```

Comment:
In general I like to avoid having exception handling to control flow, saving it for those instances where we really can't avoid it. In this case you could you find_by_id vs find to avoid raising an error. Though, looking at the code I don't think we need the host object at all, so maybe better to do `content_facet = Katello::Host::ContentFacet.find_by_host_id(host_id)`

Looks like your test will still pass with this tweak. Thanks for adding test coverage!

Reply:
I didn't know `find_by_host_id` existed, seems perfect.

Reply:
Rails magic!